packages/langchain-graph-retriever/tests/adapters/test_astra.py

```diff
@@ -107,59 +117,131 @@ def test_create_metadata_query_user() -> None:
+    keyspace = os.environ.get("ASTRA_DB_KEYSPACE", "default_keyspace")
+    api_endpoint = os.environ["ASTRA_DB_API_ENDPOINT"]
+
+    admin = AstraDBDatabaseAdmin(api_endpoint=api_endpoint, token=token)
+    admin.create_keyspace(keyspace)
+
+    # Sometimes the creation of the store fails because the keyspace isn't
+    # created yet. To avoid that, poll the list of keyspaces until we
+    # confirm it is created.
+    found = False
+    t_end = time.time() + 5  # run 5 seconds
+    while time.time() < t_end:
+        keyspaces = admin.list_keyspaces()
+        if keyspace in keyspaces:
+            found = True
+            break
+
+        print(f"Waiting for keyspace '{keyspace}'...")  # noqa: T201
+        time.sleep(0.01)
+
+    assert found, f"Keyspace '{keyspace}' not created"
+    yield _AstraConfig(token=token, keyspace=keyspace, api_endpoint=api_endpoint)
+
+    admin.drop_keyspace(keyspace)
+
+
 class TestAstraAdapter(AdapterComplianceSuite):
     @pytest.fixture(scope="class")
     def adapter(
         self,
-        enabled_stores: set[str],
         animal_embeddings: Embeddings,
         animal_docs: list[Document],
+        astra_config: _AstraConfig,
     ) -> Iterator["AstraAdapter"]:
-        if "astra" not in enabled_stores:
-            pytest.skip("Pass --stores=astra to test Astra")
-            return
-
-        import os
-
-        from astrapy import AstraDBDatabaseAdmin
-        from astrapy.authentication import StaticTokenProvider
-        from dotenv import load_dotenv
         from langchain_astradb import AstraDBVectorStore
-        from langchain_graph_retriever.adapters.astra import (
-            AstraAdapter,
+
+        store = AstraDBVectorStore(
+            embedding=animal_embeddings,
+            collection_name="animals",
+            namespace=astra_config.keyspace,
+            token=astra_config.token,
+            api_endpoint=astra_config.api_endpoint,
+            pre_delete_collection=True,
         )
+        store.add_documents(animal_docs)
 
-        load_dotenv()
+        yield AstraAdapter(store)
 
-        token = StaticTokenProvider(os.environ["ASTRA_DB_APPLICATION_TOKEN"])
-        keyspace = os.environ.get("ASTRA_DB_KEYSPACE", "default_keyspace")
-        api_endpoint = os.environ["ASTRA_DB_API_ENDPOINT"]
+        store.delete_collection()
 
-        admin = AstraDBDatabaseAdmin(api_endpoint=api_endpoint, token=token)
-        admin.create_keyspace(keyspace)
 
-        # Sometimes the creation of the store fails because the keyspace isn't
-        # created yet. To avoid that, poll the list of keyspaces until we
-        # confirm it is created.
-        found = False
-        t_end = time.time() + 5  # run 5 seconds
-        while time.time() < t_end:
-            keyspaces = admin.list_keyspaces()
-            if keyspace in keyspaces:
-                found = True
-                break
+VECTORIZE_EXPECTATION_OVERRIDES: dict[tuple[str, str], list[str]] = {
```

Comment:
Not many tests had to be overridden, so that is good -- you had a pretty close to accurate embedding function!